src/ngraph_utils.cc

```diff
@@ -164,6 +164,17 @@ const gtl::ArraySlice<DataType>& NGraphIndexDTypes() {
   return result;
 }
 
+Status CheckAxisDimInRange(std::vector<int64> axes, size_t rank) {
+  for (auto i : axes) {
+    if (i < (int)-rank || i >= (int)rank) {
+      return errors::InvalidArgument("Axis Dimension is out of range. Got ", i,
```

Comment:
`rank` is of type `size_t`, so negating it will cause an unsigned wraparound. Not sure what `(int)-rank` will actually  mean.
How about casting first before negating:
`-(static_cast<int>(rank))`

Reply:
That is what was happening earlier when there was no cast - negating was causing the wraparound and the tests were failing. With the cast I am getting the correct values for the range and the tests are passing.

Reply:
Cast it before using inside if condition, for clarity? completely optional.